posthog/cdp/templates/hubspot/template_hubspot.py

```diff
@@ -96,6 +96,242 @@
     },
 )
 
+template_event: HogFunctionTemplate = HogFunctionTemplate(
+    status="beta",
+    id="template-hubspot-event",
+    type="destination",
+    name="Hubspot",
+    description="Send events to Hubspot.",
+    icon_url="/static/services/hubspot.png",
+    category=["CRM", "Customer Success"],
+    hog="""
+if (empty(inputs.email)) {
+    print('`email` input is empty. Not sending event.')
+    return
+}
+
+let properties := {}
+
+for (let key, value in inputs.properties) {
+    if (not empty(value)) {
+        if (typeof(value) in ('object', 'array', 'tuple')) {
+            properties[key] := jsonStringify(value)
+        } else {
+            properties[key] := value
+        }
+    }
+}
+
+if (inputs.include_all_properties) {
+    for (let key, value in event.properties) {
+        if (not empty(value) and not key like '$%') {
+            if (typeof(value) in ('object', 'array', 'tuple')) {
+                properties[key] := jsonStringify(value)
+            } else {
+                properties[key] := value
+            }
+        }
+    }
+}
+
+let eventSchema := fetch(f'https://api.hubapi.com/events/v3/event-definitions/{event.event}/?includeProperties=true', {
```

Comment:
we might want to sanitize the `event.event` in the URL here and below with `encodeURLComponent`... Otherwise an event name with a `/` will cause problems for some user one of these days.

Reply:
Good point. Hubspot only allows events that start with a letter and can only contain lowercase letters, numbers, underscores, and hyphens. I've added event validation.

I don't think we should automatically remove all disallowed characters so I am just throwing an error.